org.eclipse.ui.genericeditor/src/org/eclipse/ui/internal/genericeditor/AutoEditStrategyRegistry.java

```diff
@@ -62,16 +64,24 @@ public AutoEditStrategyRegistry() {
 	 * @return the list of {@link IAutoEditStrategy} contributed for at least
 	 *         one of the content types.
 	 */
-	public List<IAutoEditStrategy> getAutoEditStrategies(ISourceViewer sourceViewer, ITextEditor editor, Set<IContentType> contentTypes) {
+	public List<IAutoEditStrategy> getAutoEditStrategies(ISourceViewer sourceViewer, ITextEditor editor,
+			Set<IContentType> contentTypes) {
 		if (this.outOfSync) {
 			sync();
 		}
-		return this.extensions.values().stream()
-			.filter(ext -> contentTypes.contains(ext.targetContentType))
-			.filter(ext -> ext.matches(sourceViewer, editor))
-			.sorted(new ContentTypeSpecializationComparator<IAutoEditStrategy>())
-			.map(GenericContentTypeRelatedExtension<IAutoEditStrategy>::createDelegate)
-			.collect(Collectors.toList());
+		return this.extensions.values().stream().filter(ext -> contentTypes.contains(ext.targetContentType))
+				.filter(ext -> ext.matches(sourceViewer, editor))
+				.sorted(new ContentTypeSpecializationComparator<IAutoEditStrategy>()) //
+				.map(ext -> {
+					IAutoEditStrategy delegateInstance = ext.createDelegate();
+					if (sourceViewer instanceof ISourceViewerExtension6
+							&& delegateInstance instanceof ITextViewerLifecycle) {
+						((ISourceViewerExtension6) sourceViewer)
+								.addTextViewerLifecycle((ITextViewerLifecycle) delegateInstance);
```

Comment:
The contributor (registry here) shouldn't have to explicitly call `viewer.addTextViewerLifecycle`, but instead the viewer should call `addTextViewerLifecycle` internally when installing the auto-edit strategies.

Let's look at TextHover for a more concrete proposal ⬇️

Reply:
Fixed